### PR TITLE
Fix: use the correct anchor in edit summary in zh.wiki

### PR DESCRIPTION
**Pull request by cooltey**
https://phabricator.wikimedia.org/T70265

**Steps to reproduce**
1. Go to https://zh.wikipedia.org/wiki/User:Cooltey/Sandbox on `zh-hant` wiki.
2. Change the language to `zh-hans`
3. Edit the last section `正體 中文 國家` and submit.
4. Go to https://zh.wikipedia.org/w/index.php?title=User:Cooltey/Sandbox&action=history to see if the edit summary contains the correct link to https://zh.wikipedia.org/wiki/User:Cooltey/Sandbox#正體_中文_國家

### CODE REVIEW

File: app/src/main/java/org/wikipedia/edit/EditSectionActivity.kt [hunk 1]
```diff
@@ -202,8 +202,8 @@ class EditSectionActivity : BaseActivity() {
     }
 
     private fun doSave(token: String) {
-        var summaryText = if (sectionHeading.isNullOrEmpty() ||
-            StringUtil.addUnderscores(sectionHeading) == pageTitle.prefixedText) "/* top */" else "/* $sectionHeading */ "
+        var summaryText = if (sectionAnchor.isNullOrEmpty() || sectionAnchor == pageTitle.prefixedText) "/* top */"
```
Comment (dbrant) [inline app/src/main/java/org/wikipedia/edit/EditSectionActivity.kt:205]: This comparison needs `addUnderscores(sectionAnchor)`

**Review by dbrant — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/wikipedia/edit/EditSectionActivity.kt (hunk 1 above)
Comment (cooltey) [reply app/src/main/java/org/wikipedia/edit/EditSectionActivity.kt:205]: The `sectionAnchor` has already contained underscores, should I still need to apply the `addUnderscores`?

Comment (dbrant) [reply app/src/main/java/org/wikipedia/edit/EditSectionActivity.kt:205]: When I try to edit an article with spaces in the title, the `sectionAnchor` still has spaces, not underscores.  And in fact when I edit an article that has a title with HTML in it, such as today's featured article (_Grevillea juniperina_), the `sectionAnchor` has HTML tags in it.

Comment (cooltey) [reply app/src/main/java/org/wikipedia/edit/EditSectionActivity.kt]: Got it. Thanks for checking. Done.

**Pull request merged**